packages/cli/src/daemon-config.ts

```diff
@@ -88,18 +88,7 @@ export class DaemonHTTPApiConfig {
   /**
    * Port to listen on.
    */
-  @jsonMember(AnyT, {
```

Comment:
Yes; the checks happen now explicitly during the config assignment (newly added) and daemon creation - I actually had it at the CLI port evaluation as well but that seemed redundant. This follows the initial pattern before I added my first fix.

Prior to @ukstv updates there was no easy/separate way of coercing the port for CLI switches besides putting it into the daemon config parser itself, but it turned out it was a bit hacky. Also, doing it during the config parser it is actually fired twice - once for instantiation and the second time for the actual launch of the daemon. Having it now explicitly check on the entry points feels a bit cleaner and establishes a pattern to do additional sanity checks on potential future values down the road too (e.g. db URL enforcement via ENV).

Reply:
It feels to me like it would make sense to check it in the two entry points for the two ways it can be set - config file and CLI flags.

The alternative would be to do it in a single place further down, in a path that all entry points call into.  This version of the PR seems to mix both - checking at the edge for CLI flags, but also further down in the CeramicDaemon constructor that all paths call into